bundles/org.openhab.voice.mimictts/src/main/java/org/openhab/voice/mimic/internal/MimicTTSService.java

```diff
@@ -223,27 +230,46 @@ public AudioStream synthesize(String text, Voice voice, AudioFormat requestedFor
             throw new TTSException("The passed AudioFormat is unsupported");
         }
-        String encodedText;
-        try {
-            encodedText = URLEncoder.encode(text, StandardCharsets.UTF_8.toString());
-        } catch (UnsupportedEncodingException e) {
-            throw new IllegalArgumentException("Cannot encode text in URL " + text);
-        }
 
         String ssml = "";
         if (text.startsWith("<speak>")) {
             ssml = "&ssml=true";
         }
 
-        // create the audio byte array for given text, locale, format
-        String urlTTS = config.url + SYNTHETIZE_URL + "?text=" + encodedText + "&voice="
-                + ((MimicVoice) voice).getTechnicalName() + ssml + "&noiseScale=" + config.audioVolatility + "&noiseW="
-                + config.phonemeVolatility + "&lengthScale=" + config.speakingRate + "&audioTarget=client";
+        // create the url for given locale, format
+        String urlTTS = config.url + SYNTHETIZE_URL + "?voice=" + ((MimicVoice) voice).getTechnicalName() + ssml
+                + "&noiseScale=" + config.audioVolatility + "&noiseW=" + config.phonemeVolatility + "&lengthScale="
+                + config.speakingRate + "&audioTarget=client";
         logger.debug("Querying mimic with URL {}", urlTTS);
-        RawType responseWav = HttpUtil.downloadData(urlTTS, "audio/wav", false, -1);
-        if (responseWav == null) {
-            throw new TTSException("Cannot get wav from mimic url " + urlTTS);
+
+        // prepare the response as an inputstream
+        InputStreamResponseListener inputStreamResponseListener = new InputStreamResponseListener();
+        // we will use a POST method for the text
+        StringContentProvider textContentProvider = new StringContentProvider(text, StandardCharsets.UTF_8);
+        httpClient.POST(urlTTS).content(textContentProvider).accept("audio/wav").send(inputStreamResponseListener);
+
+        // compute the estimated timeout using a "stupid" method based on text length, as the response time depends on
+        // the requested text. Average speaker speed estimated to 10/second.
+        // Will use a safe margin multiplicator (x5) to accept very slow mimic server
+        // So the constant chosen is 5 * 10 = /2
+        int timeout = text.length() / 2;
+
+        // check response status and return AudioStream
+        Response response;
+        try {
+            response = inputStreamResponseListener.get(timeout, TimeUnit.SECONDS);
+            if (response.getStatus() == 200) {
+                String lengthHeader = response.getHeaders().get(HttpHeader.CONTENT_LENGTH);
+                long length = Long.parseLong(lengthHeader);
+                return new InputStreamAudioStream(inputStreamResponseListener.getInputStream(), AUDIO_FORMAT, length);
+            } else {
+                String errorMessage = "Cannot get wav from mimic url " + urlTTS + " with HTTP response code "
```

Comment:
Looking what I have done in the remote openHAB binding, I was aborting the request when code was not OK,, something like:
```
response.abort(new Exception(response.getReason()));
```

Reply:
It seems like a good practice, especially with callback response / deferred inputstream. Done

TY for your review !